src/main/java/io/fabric8/jenkins/openshiftsync/OpenShiftUtils.java

```diff
@@ -84,6 +91,18 @@ public synchronized static OpenShiftClient getOpenShiftClient() {
     return openShiftClient;
   }
 
+  // Get the current OpenShiftClient and configure to use the current Oauth token.
+  public synchronized static OpenShiftClient getAuthenticatedOpenShiftClient() {
+    if(openShiftClient != null){
+      String token = CredentialsUtils.getCurrentToken();
+      if(token.length() > 0) {
+        openShiftClient.getConfiguration().setOauthToken(token);
```

Comment:
you've refactored this code to the point where it's not obvious to me that getOpenShiftClient() can't just do this work now...   is there still a recursive flow if it did so?